### ISSUE TITLE
Differences in OT and data assembly information

### ISSUE THREAD

**Issue by komalsrathi** [d3b-center/d3b-bixu-data-assembly#14]
Currently, the generalized patient reporting code cannot be automated because for each new patient report, I have manually check and map all missing information between various file sources i.e. KF clinical data (new patients are not in OT), Open Targets and Data assembly histology file. 

Here are differences in some columns between OT histology file and data assembly file (To be discussed in toolkit as well):

```
# minimal set of columns to be used for pediatric samples within the generalized reporting code 
hist_cols_used <- c("Kids_First_Biospecimen_ID","sample_id","cohort","cohort_participant_id",
                    "parent_aliquot_id","aliquot_id",
                    "experimental_strategy","RNA_library",
                    "sample_type","tumor_descriptor",
                    "pathology_diagnosis","integrated_diagnosis",
                    "short_histology","broad_histology","molecular_subtype",
                    "OS_days","OS_status")

# read data assembly histology and subset to minimal columns
# add parent_aliquot_id as this field is not present in the data assembly file
data_assembly <- read.delim('~/Downloads/histologies-add-bixops-uptoPNOC008-52.tsv')
data_assembly <- data_assembly %>%
  filter(cohort %in% c("PNOC", "CBTN")) %>%
  mutate(parent_aliquot_id = NA) %>%
  dplyr::select(hist_cols_used) # 3687

# read open targets and subset to minimal columns
open_targets <- read.delim('~/Projects/PediatricOpenTargets/OpenPedCan-analysis/data/histologies.tsv')
open_targets <- open_targets %>%
  dplyr::filter(cohort == "PBTA") %>%
  dplyr::select(hist_cols_used) # 2984

# subset both files on common samples
common_samples <- intersect(data_assembly$Kids_First_Biospecimen_ID, open_targets$Kids_First_Biospecimen_ID)
data_assembly <- data_assembly %>%
  filter(Kids_First_Biospecimen_ID %in% common_samples) %>%
  arrange(Kids_First_Biospecimen_ID)
open_targets <- open_targets %>%
  filter(Kids_First_Biospecimen_ID %in% common_samples) %>%
  arrange(Kids_First_Biospecimen_ID)

# replace NA to blank so we can compare the columns in both histology files
data_assembly[is.na(data_assembly)] <- "" 
open_targets[is.na(open_targets)] <- ""

# check differences in the minimal set of columns only
for(i in 1:length(hist_cols_used)){
  col_to_check <- hist_cols_used[i]
  diff_rows <- open_targets[which(open_targets[,col_to_check] != data_assembly[,col_to_check]),] %>%
    nrow() 
  print(paste(col_to_check, diff_rows, sep = ": "))
}

[1] "Kids_First_Biospecimen_ID: 0"
[1] "sample_id: 0"
[1] "cohort: 2984"
[1] "cohort_participant_id: 0"
[1] "parent_aliquot_id: 2840"
[1] "aliquot_id: 0"
[1] "experimental_strategy: 0"
[1] "RNA_library: 0"
[1] "sample_type: 0"
[1] "tumor_descriptor: 94"
[1] "pathology_diagnosis: 80"
[1] "integrated_diagnosis: 1364"
[1] "short_histology: 2111"
[1] "broad_histology: 2111"
[1] "molecular_subtype: 1615"
[1] "OS_days: 472"
[1] "OS_status: 0"
```

So, in short:
1) `parent_aliquot_id` is absent in data assembly
2) there are many differences in cohort (for e.g OT uses `PBTA` and data assembly uses `PNOC/CBTN`), tumor_descriptor, pathology_diagnosis, integrated_diagnosis, short_histology, broad_histology, molecular_subtype, OS_days

cc: @aadamk

**Comment by komalsrathi** [comment]
Actually just realized there are empty values and Not Reported in data assembly (DA) and only empty values in OT so I fixed that manually as well and found these 13 differences:

Kids_First_Biospecimen_ID | OT | DA
-- | -- | --
BS_264RCK7G | 1756 | 2802
BS_2EAATK5V | 6585 | 7090
BS_382HZBG4 | 1839 | 2714
BS_6NRTJSYK | 1839 | 2714
BS_6P50QTPP | 6585 | 7090
BS_92MT680S | 1839 | 2714
BS_C8FMN2ZR | 1756 | 2802
BS_DW1CYEXP | 6585 | 7090
BS_E5B2YZJH | 1756 | 2802
BS_ERMGD2N5 | 6585 | 7090
BS_JC8YV5V9 | 1839 | 2714
BS_N9BF56FP | 1839 | 2714
BS_W3R88RFP | 6585 | 7090

We should also think about having a consistent value where data is Not available, Not applicable....

**Comment by komalsrathi** [comment]
[OT_vs_DA_qc.xlsx](https://github.com/d3b-center/d3b-bixu-data-assembly/files/8623687/OT_vs_DA_qc.xlsx)

Attached is a comparison of the fields (see sheet name) that don't match up either because of missing data or discrepancies.

**Comment by jharenza** [comment]
@komalsrathi can you do one more check - if you pull the base histologies file today, are these still missing? if so, then we will have to talk with @nicholasvk about that

**Comment by komalsrathi** [comment]
[OT_vs_ADAPT_qc.xlsx](https://github.com/d3b-center/d3b-bixu-data-assembly/files/8624701/OT_vs_ADAPT_qc.xlsx)

@jharenza I just pulled the file and here is the comparison.

**Comment by jharenza** [comment]
thanks @komalsrathi - this was a previous issue that was fixed but looks to be back - cc @nicholasvk and added a ticket [here](https://github.com/PediatricOpenTargets/ticket-tracker/issues/306)

**Comment by aadamk** [comment]
given that ADAPT regularly updates the base histologies file with IDs, path_Dx, path_free_text_dx, i'm thinking the best plan going forward is: 

1. rename `cohort` to `internal_cohort` in the bix master histologies file based on @jharenza's suggestion.
2. continue the current bix processing for ploidy, purity, and germline sex estimate in the bix master histologies file 
3. get the short, broad, and molecular subtype annotation cwl into production (Matt and Alex are working on this), which would create an id + subtype + short/broad hist file. 
4. hand over the bix-processed fields from (2) (essentially an id-to-ploidy,purity,sex_estimate mapping) + the id-to-subtype annotation file from (3) to @nicholasvk, who can then put this all together into a processed histologies view. 

That way, we can leave the cwl processing from Bo as is, the base histology processing as is (which I believe will contain all OT studies soon - @jharenza can you inform here?), and then have an ETL to add on the additional data above to produce the processed file. 

Let me know your thoughts @komalsrathi @jharenza @zhangb1 @yuankunzhu @nicholasvk

**Comment by jharenza** [comment]
>rename cohort to internal_cohort in the bix master histologies file based on @jharenza's suggestion.

One note on this - we had decided to make a separate table of `internal_cohort` just so those who use that field are _intentional_, as it will contain sensitive clinical trial numbers. We will convert `cohort == "CBTN | PNOC"` --> `cohort == "PBTA"`

>That way, we can leave the cwl processing from Bo as is, the base histology processing as is (which I believe will contain all OT studies soon - @jharenza can you inform here?), and then have an ETL to add on the additional data above to produce the processed file.

This is already integrated (TCGA, TARGET, GTEX), though we need to update some DGD diagnosis mappings.

>get the short, broad, and molecular subtype annotation cwl into production (Matt and Alex are working on this), which would create an id + subtype + short/broad hist file.

This is something still pretty ongoing in OpenPedCan, so be cautious that it will need updates in the toolkit before totally accurate

**Comment by komalsrathi** [comment]
Closing as the patient report is now completely reliant on the data assembly file and does not use OT histologies. If there are any issues with the data assembly, I will open a new issue.
